storage/storage.go

```diff
@@ -267,7 +271,9 @@ func (storage DBStorage) WriteReportForCluster(
 	if err != nil {
 		return err
 	}
-	defer statement.Close()
+	defer func() {
+		err = statement.Close()
+	}()
```

Comment:
In this case, `err` is a named return value of `WriteReportForCluster` function. The deferred `func` will be executed, as usual, at the end of the function execution. In the case that `err` takes some value in the deferred function execution, it will be returned to the caller.
AFAIK, if we are not using the `err` variable, probably the checkers should fail

Reply:
Ok, Thanks, I had to ask Pavel, but now I got it. I understand named returns, but I got confused by the missing return statement - which cant be there since it's deferred... :D 

I think this isnt obvious right away, though, maybe worthy of a comment? But thats just me..

Reply:
@Bee-lee This is a pretty wild use of named returns to be fair. It's one of the edge cases suitable for interview questions - "What does this function really returns" :wink:

Reply:
You already know, I'm a borderline :-D